pybop/costs/fitting_costs.py

```diff
@@ -39,10 +39,23 @@ def _evaluate(self, x, grad=None):
         """
         prediction = self.problem.evaluate(x)
 
-        if len(prediction) < len(self._target):
-            return np.float64(np.inf)  # simulation stopped early
+        for key in prediction:
+            if key not in ["Time [s]", "Discharge capacity [A.h]"]:
+                if len(prediction.get(key, [])) != len(self._target.get(key, [])):
+                    return np.float64(np.inf)  # prediction doesn't match target
+
+        e = np.array(
+            [
+                np.sqrt(np.mean((prediction[signal] - self._target[signal]) ** 2))
+                for signal in prediction
+                if signal not in ["Time [s]", "Discharge capacity [A.h]"]
```

Comment:
We don't need the additional variables in error computation. Likewise to above, filtering based on the additional variables and `Time [s]` might be a clearer choice.

Reply:
I think that users should have free choice of variables to include in `signal`. We would have to add extra warnings if we are to disallow some variables.